packages/use-contractkit/src/ethers.ts

```diff
@@ -6,10 +6,10 @@ import { useContractKit } from './use-contractkit';
 import { useIsMounted } from './utils/useIsMounted';
 
 export const useProvider = (): Web3Provider => {
-  const { kit } = useContractKit();
+  const { kit, network } = useContractKit();
   const provider = kit.web3.currentProvider as unknown as ExternalProvider;
   return useMemo(() => {
-    return new Web3Provider(provider);
+    return new Web3Provider(provider, {chainId: network.chainId, name: network.name});
```

Comment:
Looks good but like linter says you'll need to update the memo list too

Reply:
yep.

Reply:
was wondering about your thoughts on moving the provider up into the App Context Provider. Or maybe its own ProviderProvider Context? so

Reply:
Your comment cuts out. What would be the benefit of the move?

Reply:
One shared instance rather than each component that uses this having its own